tests/appsec/test_asm_standalone.py

```diff
@@ -0,0 +1,198 @@
+from utils import weblog, interfaces, scenarios, features
+from utils._context.header_tag_vars import *
+import time
+
+
+@scenarios.appsec_standalone
+class Test_AppSecPropagation:
+    """APM correctly propagates AppSec events in distributing tracing."""
+
+    def setup_trace_no_upstream_no_attack_is_kept_by_rate_limiter_minimum(self):
+        trace_id = 1212121212121212121
+        parent_id = 34343434
+        # Ensure this is the first span received in 60 seconds
+        time.sleep(60)
```

Comment:
If there is no easy solution, what I did in my smoke tests was split this in two test:

- One only checks the fact that only one trace per minute is keep by the sampler 
- The other is the one that checks everything, and instead of assert a fixed sampling priority it checks that is set b de sampler (AUTO_DROP[0] or AUTO_KEEP[1])

Reply:
you want to wait for a first span to be seen, right ? 

You can do that : 

```python
# wait for a very first /trace before sending request
interfaces.library.wait_for(lambda data: data["path"] != "/v0.4/traces")
```

Reply:
Not exactly, we have a rate limiter set to keep 1 trace per minute, so I wanted to make sure that's the trace kept (being the first one received in 60 seconds)